packages/jmix-front-generator/src/common/studio/studio-model.ts

```diff
@@ -3,37 +3,111 @@ import {RestService, RestServiceMethod} from "../model/cuba-model";
 
 export const enum StudioTemplatePropertyType {
   ENTITY = 'ENTITY',
-  VIEW = 'VIEW',
-  NESTED_ENTITY_VIEW = 'NESTED_ENTITY_VIEW',
   STRING = 'STRING',
   BOOLEAN = 'BOOLEAN',
   INTEGER = 'INTEGER',
   OPTION = 'OPTION',
-  MULTI_OPTION = 'MULTI_OPTION',
-  REST_QUERY = 'REST_QUERY',
-  REST_SERVICE_METHOD = 'REST_SERVICE_METHOD',
   POLYMER_COMPONENT_NAME = 'POLYMER_COMPONENT_NAME',
   PASSWORD = 'PASSWORD',
   GRAPHQL_QUERY = 'GRAPHQL_QUERY',
   MENU_ITEM = 'MENU_ITEM',
   ATTRIBUTE = 'ATTRIBUTE',
   ATTRIBUTES_ARRAY = 'ATTRIBUTES_ARRAY',
+
+  // deprecated types
+  MULTI_OPTION = 'MULTI_OPTION',
+  VIEW = 'VIEW',
+  NESTED_ENTITY_VIEW = 'NESTED_ENTITY_VIEW',
+  REST_QUERY = 'REST_QUERY',
+  REST_SERVICE_METHOD = 'REST_SERVICE_METHOD',
 }
 
-export interface StudioTemplateProperty {
+export interface BaseStudioTemplateProperty {
   code: string;
   caption: string;
   propertyType: StudioTemplatePropertyType;
   defaultValue?: string;
   required?: boolean;
-  relatedProperty?: string;
-  options?: string[];
   step?: StepQuestionParam;
   // --not supported
   //advanced: boolean;
   //filterScript: string;
 }
 
+export interface EntityStudioTemplateProperty extends BaseStudioTemplateProperty {
+  propertyType: StudioTemplatePropertyType.ENTITY;
+}
+
+export interface StringStudioTemplateProperty extends BaseStudioTemplateProperty {
+  propertyType: StudioTemplatePropertyType.STRING;
+}
+
+export interface BooleanStudioTemplateProperty extends BaseStudioTemplateProperty {
+  propertyType: StudioTemplatePropertyType.BOOLEAN;
+}
+
+export interface IntegerStudioTemplateProperty extends BaseStudioTemplateProperty {
+  propertyType: StudioTemplatePropertyType.INTEGER;
+}
+
+export interface OptionStudioTemplateProperty extends BaseStudioTemplateProperty {
+  propertyType: StudioTemplatePropertyType.OPTION;
+  options: string[];
+}
```

Comment:
I'd suggest to revert this change for now, I'm not 100% sure it won't break anything in Studio.

Reply:
I didn't change existing logic, just define typescript type for every studio property inside the generator. I thought that Studio don't see typescript types.